packages/core/components/NumberRangePicker/index.tsx

```diff
@@ -50,10 +51,11 @@ export default function NumberRangePicker(props: NumberRangePickerProps) {
         extractValuesFromRangeOperatorFilterString(currentRange?.value).maxValue ?? overallMax
     );
 
+    // Instead of removing filter completely, reset to min and max and submit
     function onResetSearch() {
-        onReset();
-        setSearchMinValue("");
-        setSearchMaxValue("");
+        setSearchMinValue(overallMin);
+        setSearchMaxValue(overallMax);
+        onSearch(`RANGE(${overallMin},${overallMax})`);
```

Comment:
Debatable if this is in the scope of this PR but I think its odd we pass a string version of a range filter to the parent component rather than something like `onSearch(min, max)`

Reply:
Agreed, although onSearch is a pre-existing function also used for searchboxes so would need some additional changes to accept other args. `RANGE` as a filter value in general could use refactoring, we have to do some additional processing in number-formatter and date-formatter as well